tracer/build/_build/Build.VariableGenerations.cs

```diff
@@ -0,0 +1,263 @@
+                {
+                    foreach (var testDescription in testDescriptions)
+                    {
+                        matrix.Add(
+                            $"{explorationTestUseCase}_{testDescription.Name}",
+                            new { explorationTestUseCase, explorationTestName = testDescription.Name });
+                    }
+                }
+
+                Logger.Info($"Exploration test windows matrix");
+                Logger.Info(JsonConvert.SerializeObject(matrix, Formatting.Indented));
+                AzurePipelines.Instance.SetVariable("exploration_tests_windows_matrix", JsonConvert.SerializeObject(matrix, Formatting.None));
+            }
+
+            void GenerateExplorationTestsLinuxMatrix(IEnumerable<string> useCases)
+            {
+                var testDescriptions = ExplorationTestDescription.GetAllExplorationTestDescriptions();
+                var targetFrameworks = TargetFramework.GetFrameworks(new[] { TargetFramework.NET461, TargetFramework.NETSTANDARD2_0, });
+
+                var baseImages = new[] { "debian", "alpine" };
+
+                var matrix = new Dictionary<string, object>();
+
+                foreach (var baseImage in baseImages)
+                {
+                    foreach (var explorationTestUseCase in useCases)
+                    {
+                        foreach (var targetFramework in targetFrameworks)
+                        {
+                            foreach (var testDescription in testDescriptions)
+                            {
+                                if (testDescription.IsFrameworkSupported(targetFramework))
+                                {
+                                    matrix.Add(
+                                        $"{baseImage}_{targetFramework}_{explorationTestUseCase}_{testDescription.Name}",
+                                        new { baseImage, targetFramework, explorationTestUseCase, explorationTestName = testDescription.Name });
+                                }
+                            }
+                        }
+                    }
+                }
+
+
+                Logger.Info($"Exploration test linux matrix");
+                Logger.Info(JsonConvert.SerializeObject(matrix, Formatting.Indented));
+                AzurePipelines.Instance.SetVariable("exploration_tests_linux_matrix", JsonConvert.SerializeObject(matrix, Formatting.None));
+            }
+        };
+}
+
+[Flags]
+public enum GenerateMatricesTarget
+{
+    none = 0,
+    integration_tests_windows_matrix = 1 << 0,
+    integration_tests_windows_iis_matrix = 1 << 1,
+    integration_tests_linux_matrix = 1 << 2,
+    exploration_tests_windows_matrix = 1 << 3,
+    exploration_tests_linux_matrix = 1 << 4,
+
+    integration_tests_windows_matrices = integration_tests_windows_matrix | integration_tests_windows_iis_matrix,
+    exploration_tests_matrices = exploration_tests_windows_matrix | exploration_tests_linux_matrix
```

Comment:
As per the other comments, I think we can remove this entirely and always build all matrices.